test-interface/src/main/scala/org/scalajs/testinterface/ScalaJSClassLoader.scala

```diff
@@ -6,12 +6,19 @@ import java.net.URL
 import java.io.InputStream
 import java.util.Enumeration
 
-/** A dummy [[java.lang.ClassLoader]] that allows to store a JavaScript object
- *  against which classes are resolved. The only reason it extends
- *  [[java.lang.ClassLoader]] is typing.
+/** A dummy [[java.lang.ClassLoader]] for Scala.js testing frameworks.
+ *
+ *  This class loader does *not* implement the contract of
+ *  [[java.lang.ClassLoader]]. It is exclusively used as a dummy class loader
+ *  to preserve source compatibility with the sbt testing interface.
  */
-final class ScalaJSClassLoader(
-    val namespace: js.Dynamic) extends ClassLoader(null) {
+final class ScalaJSClassLoader extends ClassLoader(null) {
+
+  @deprecated(
+      "The `namespace` argument is ignored. " +
+      "Use the constructor with 0 argument.",
+      "1.0.0")
+  def this(namespace: js.Dynamic) = this()
```

Comment:
We can't deprecate it in 0.6.x because it's *necessary* in 0.6.x.

My idea would be to keep it deprecated during 1 or 2 milestones, and then actually remove it before the RCs. More generally, we'll need to make a large swipe of all deprecated methods in the codebase before the RCs.